LUMO_toolbox/DOTHUB_LUMOdataQualityCheck.m

```diff
@@ -58,9 +64,91 @@ function DOTHUB_dataQualityCheck(nirsFileName,printFigFlag)
-% - this script does not focus on motion content, but providing
+%timeSelectFlag allows the user to select a clean period (motion free) for analysis. 
+%A subset of channels are plotted to decrease computer load (this can
+%always be changed to plot all channels).
+if (timeSelectFlag==1) 
+    my_ans = 'n';
+    while my_ans == 'n';
+    fig1 = figure; set(fig1, 'units', 'normalized', 'outerposition', [0 0 0.8 0.8], 'color', [1 1 1])
+    plot(d(:,1500:2000)); xlim([0 size(d,1)]); ylim([0 1.5]); title('Select segment to include (start-end pair) - Press enter when finished', 'fontsize', 20)
+    xlabel ('Time (samples)','Fontsize', 15); ylabel('Intensity (raw)', 'Fontsize', 15)
+    set(gca, 'Yscale', 'log')
+    hold on
+% do data brushing manually
+    [x, ~] = ginput;
+% If you select segments
+    if  ~isempty(x)
+        % Repeat if x data is not an even number
+        if  mod(size(x,1),2) ~= 0
+            close all; clc
+            warning('Select start-end of each segments, should be pairs of numbers (even)')
+            clear x
+            continue
+        end
+        % If first selected point is before 0 assume 1
+        if x(1)<0
+            x(1) = 1;
+        end
+        % If last selected point is bigger than size data, assume is the last point
+        if x(end)>size(d,1)
+            x(end) = size(d,1);
+        end
+  
+        mrk_data = zeros(size(x,1)/2,2);
+        mrk_data(:,1) = ceil(x(1:2:size(x,1)));
+        mrk_data(:,2) = ceil(x(2:2:size(x,1)));
+        
+% Check outcome
+        close all
+        fig1 = figure; set(fig1, 'units', 'normalized', 'outerposition', [0 0 0.8 0.8], 'color', [1 1 1])
+        plot(d(:,1500:2000)); xlim([0 size(d,1)]); title('Select segment to include (start-end pair) - Press enter when finished', 'fontsize', 14)
+        hold on
+        for i = 1: size(mrk_data,1)
+            area([mrk_data(i,1) mrk_data(i,2)], [1.5 1.5], 'FaceColor', 'c',...
+                'LineStyle', 'none'); alpha 0.2
+            hold on
+        end
+        xlabel ('Time (samples)','Fontsize', 15);ylabel('Optical density', 'Fontsize', 15)
+        set(gca,'fontsize',15); set(gca, 'Yscale', 'log')
+        
+        % Ask if ok and continue, or not ok and repeat selection
+        prompt = {'Selection correct? Yes = y , No = n '};
+        dlgtitle = 'Input';
+        dims = 1;
+        definput = {'0','hsv'};
+        my_ans = inputdlg(prompt,dlgtitle,dims,definput);
+        my_ans = my_ans{1};
+        
+        if my_ans == 'n'
+            clear x mrk_data
+            close all
+            clc
+            warning('Repeat selection')
+            continue
+        end
+        
+       clean_idx = zeros(size(d,1),1);
+       for i = 1: size(mrk_data,1)
+           clean_idx(mrk_data(i,1):mrk_data(i,2)) = 1;
+       end
+        % if noisy segments are not selected
+   else
+       clean_idx = ones(size(d,1),1);
+       my_ans = 'y';
+  end
+   data.ini = mrk_data(1,1);
```

Comment:
Do you need the data variable? Seems unnecessary

Reply:
Took this out